### PR TITLE
Feature 732

### PR DESCRIPTION
**Pull request by fjaguado**
# PR Details

A new directive for testing purpose has been added. Furthermore a new config method for the library has been added to be able to config if the productionMode is enabled or not.

## Related Issue

#732  
https://github.com/systelab/systelab-components-wdio-test/issues/37

## Motivation and Context

The idea is to be able to enable or disable the 'data-test-id' HTML attributes depending on whether the production mode is enabled or not. Furthermore adds a common to work with frontend test.

## How Has This Been Tested

A new test file has been added to the library.

## Types of changes

- [ x ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ x ] I have read the **CONTRIBUTING** document
- [ x ] My code follows the code style of this project
- [ x ] My change requires a change to the documentation 
- [ x ] I have updated the documentation accordingly (README.md for each UI component)
- [ x ] I have added tests to cover my changes (at least 1 spec for each UI component with the same coverage as the master branch)
- [ x  ] All new and existing tests passed
- [ x ] A new branch needs to be created from master to evolve previous versions
- [ x ] Increase version in package.json following [Semantic Versioning](https://semver.org/)
- [ x ] All UI components must be added into the showcase (at least 1 component with the default settings)
- [ x ] Add the issue into the right [project](https://github.com/systelab/systelab-components/projects) with the proper status (In progress)

### CODE REVIEW

File: projects/systelab-components/src/lib/directives/README.md [hunk 1]
```diff
@@ -45,4 +45,16 @@ i.e: when locale is spanish, if we click the '.' key it translates the '.' to ',
 
 ```html
 <input type="text" id="ID_DOSE"  systelabNumPadDecimalNumericDirective/>
-```
\ No newline at end of file
+```
+
+## testId
+
+This directives adds a HTML data-test-id attribute if the configuration is not set to production. This data field can be used
```
Comment (Franek75) [inline projects/systelab-components/src/lib/directives/README.md:52]: typo

File: projects/systelab-components/src/lib/directives/README.md [hunk 2]
```diff
@@ -45,4 +45,16 @@ i.e: when locale is spanish, if we click the '.' key it translates the '.' to ',
...
+to select the element for testing purpose.
+
+This directive also avoid to have non needed code when the
```
Comment (Franek75) [inline projects/systelab-components/src/lib/directives/README.md:55]: typo

File: projects/systelab-components/src/lib/systelab-components.module.ts [hunk 3]
```diff
@@ -294,4 +298,21 @@ import { NumpadDecimalNumericDirective } from './directives/numpad-decimal-numer
...
+			throw new Error('SystelabComponentsModule is already loaded. Please add it in AppModule only.');
+		}
+	}
+
+	public static forRoot(conf?: AppConfig): ModuleWithProviders<SystelabComponentsModule> {
+		return {
+			ngModule: SystelabComponentsModule,
+			providers: [
+				{
+					provide: APP_CONFIG, useValue: conf
```
Comment (joaquimvila) [inline projects/systelab-components/src/lib/systelab-components.module.ts:313]: What if configuration is not defined? Shouldn't we provide a default one?
I would suggest to set "productionMode" to true by default.

File: projects/systelab-components/src/lib/directives/test-id.directive.spec.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+      imports: [BrowserModule],
+      declarations: [ SystelabTestIdDirectiveTestComponent, TestIdDirective ],
+      schemas: [ CUSTOM_ELEMENTS_SCHEMA ],
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(SystelabTestIdDirectiveTestComponent);
+    fixture.detectChanges();
+  });
+
+  it('should create an instance', fakeAsync(() => {
```
Comment (joaquimvila) [inline projects/systelab-components/src/lib/directives/test-id.directive.spec.ts:26]: I would better say `should add "data-test-id" attribute with defined label into DOM element`

File: projects/systelab-components/src/lib/directives/test-id.directive.spec.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    await TestBed.configureTestingModule({
+      imports: [BrowserModule],
+      declarations: [ SystelabTestIdDirectiveTestComponent, TestIdDirective ],
+      schemas: [ CUSTOM_ELEMENTS_SCHEMA ],
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(SystelabTestIdDirectiveTestComponent);
+    fixture.detectChanges();
+  });
+
```
Comment (joaquimvila) [inline projects/systelab-components/src/lib/directives/test-id.directive.spec.ts:25]: I would have 3 different scenarios for the unit tests of this directive:
- No configuration of production mode -> DOM attribute NOT added
- Production mode = true -> DOM attribute NOT added
- Production mode = false -> DOM attribute added

**Review by joaquimvila — CHANGES_REQUESTED** [review]
Good job @fjaguado with this new directive. However, requested 2 things that I think that are important prior to merging this PR:
- Handling of default value for production mode
- Missing unit tests of the directive

File: projects/systelab-components/src/lib/directives/test-id.directive.spec.ts (hunk 4 above)
Comment (fjaguado) [reply projects/systelab-components/src/lib/directives/test-id.directive.spec.ts]: Done

File: projects/systelab-components/src/lib/directives/test-id.directive.spec.ts (hunk 5 above)
Comment (fjaguado) [reply projects/systelab-components/src/lib/directives/test-id.directive.spec.ts:72]: Done

**Review by joaquimvila — DISMISSED** [review]
Agree with implementation now. Thanks @fjaguado for the effort on this new directive!

**Review by RubKrmon — DISMISSED** [review]

**Review by RubKrmon — APPROVED** [review]

**Review by joaquimvila — APPROVED** [review]

**Review by joaquimvila — APPROVED** [review]

**Comment by Franek75** [comment]
Deeply reviewed and finally approved

**Pull request merged**